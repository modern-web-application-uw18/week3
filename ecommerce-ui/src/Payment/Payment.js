import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Payment extends Component {
    render() {
        if ( this.props.payment.description === "" ) {
            return(
                <div> ${this.props.payment.cost} per night </div>
            )
        }
        else {
            return (
                <div> ${this.props.payment.cost} per night &middot; {this.props.payment.description} </div>
            )};
        }
}

Payment.propTypes = {
    cost: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired
}

Payment.defaultProps = {
    cost: 150,
    description: "None"
}

export default Payment;