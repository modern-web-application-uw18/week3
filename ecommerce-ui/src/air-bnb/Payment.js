import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Payment extends Component {

    render() {

        const {
            cost,
            description
        } = this.props.payment;

        const hasDescription = () => {
            if (description) {
                return (
                    <span> &middot; {description}</span>
                )
            } else {
                return null;
            }
        }

        return (
            <div>${cost} per night {hasDescription()}</div>
        );
    }
}

Payment.propTypes = {
    cost: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired
};

Payment.defaultProps = {
    cost: 0,
    description: 'None provided'
};

export default Payment;
