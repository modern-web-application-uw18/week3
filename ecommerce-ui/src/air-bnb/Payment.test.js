import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Payment from './Payment';

describe('Payment', () => {
    it('renders without crashing', () => {
        const payment = {"cost": "145", "description": "Free cancellation"};
        const div = document.createElement('div');
        ReactDOM.render(<Payment payment={payment} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with cost and description', () => {
        const payment = {"cost": "200", "description": "One night cancellation fee"};
        const component = TestRenderer.create(
            <Payment payment={payment} />);
            expect(component.toJSON()).toMatchSnapshot();
    });
});