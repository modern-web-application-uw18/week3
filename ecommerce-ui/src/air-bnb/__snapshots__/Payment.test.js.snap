// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should render with cost and description 1`] = `
<div>
  $
  200
   per night 
  <span>
     Â· 
    One night cancellation fee
  </span>
</div>
`;
